plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -601,13 +619,27 @@ public ColumnHandle getDeleteRowIdColumnHandle(ConnectorSession session, Connect
                 icebergTable.location()));
     }
 
+    private Optional<ConnectorTableExecuteHandle> getTableHandleForVacuum(ConnectorSession session, IcebergTableHandle tableHandle, Map<String, Object> executeProperties)
+    {
+        Duration retentionThreshold = (Duration) executeProperties.get("retention_threshold");
```

Comment:
Should we have a safety minimum on this value? The Databricks vacuum command has a limit of 7 days, but you can enable a retention period smaller than that by opting in with a session property.

Reply:
Good point. We should do as we did in Delta Lake's vacuum. I think i implemented safety session toggle to unlock short thresholds.

Reply:
I see you added validation in `executeVacuum`. Why postpone. We can validate already here.

Reply:
I tried to follow the implementation of VACUUM for delta lake